.gitignore

```diff
@@ -1,33 +1,75 @@
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio, WebStorm and Rider
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
+
+# Generated files
+.idea/**/contentModel.xml
+
+# Sensitive or high-churn files
+.idea/**/dataSources/
+.idea/**/dataSources.ids
+.idea/**/dataSources.local.xml
+.idea/**/sqlDataSources.xml
+.idea/**/dynamic.xml
+.idea/**/uiDesigner.xml
+.idea/**/dbnavigator.xml
+
+# Gradle
+.idea/**/gradle.xml
+.idea/**/libraries
+
+# Gradle and Maven with auto-import
+# When using Gradle or Maven with auto-import, you should exclude module files,
+# since they will be recreated, and may cause churn.  Uncomment if using
+# auto-import.
+.idea/artifacts
+.idea/compiler.xml
+.idea/jarRepositories.xml
+.idea/modules.xml
+.idea/*.iml
+.idea/modules
 *.iml
-.idea/
```

Comment:
OMG how do you know all those paths to exclude or not? I feel I did this a million times and this is the best I could come up with :/

Reply:
In my experience with IntelliJ I always ignore the entire `.idea/` directory - otherwise other files end up popping up in version control.

Reply:
Oh, I just decided to find the best practices for .gitignore!

I would agree if you ask me to remove the whole `.idea` from version control. Right now, I have only two files under git control (misc.xml and vcs.xml) and both look to be useless.

Let me do this: I can first ignore the whole `.idea` and we can change that later in another pull request.

My primary goal is to allow us having the `input` subdirectory at `core/src/main/java/org/mobilitydata/gtfsvalidator/input/`

Reply:
👍